rotkehlchen/chain/evm/decoding/gearbox/gearbox_cache.py

```diff
@@ -0,0 +1,327 @@
+            )) is None:
+                continue
+            if len(farming_pool_addresses := globaldb_get_general_cache_keys_and_values_like(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_FARMING_TOKENS, pool_address),
+            )) == ZERO:
+                continue
+            if len(token_lp_addresses := globaldb_get_general_cache_keys_and_values_like(
+                cursor=cursor,
+                key_parts=(CacheType.GEARBOX_POOL_LP_TOKENS, pool_address),
+            )) == ZERO:
+                continue
+            pools[pool_address] = GearboxPoolData(
+                pool_address=pool_address,
+                pool_name=pool_name,
+                underlying_token=string_to_evm_address(underlying_token),
+                farming_pool_addresses={addr[1] for addr in farming_pool_addresses},
+                lp_token_addresses={addr[1] for addr in token_lp_addresses},
+            )
+
+    return (pools,)
+
+
+def ensure_gearbox_tokens_existence(
+        evm_inquirer: 'EvmNodeInquirer',
+        all_pools: list[GearboxPoolData],
+) -> list[GearboxPoolData]:
+    """This function receives data about gearbox pools and ensures that lp tokens and pool coins
+    exist in rotki's database."""
+    verified_pools = []
+    for pool in all_pools:
+        # Ensure pool coins exist in the globaldb.
+        # ensure underlying token exists
+        try:
+            get_or_create_evm_token(
+                userdb=evm_inquirer.database,
+                evm_address=pool.underlying_token,
+                chain_id=evm_inquirer.chain_id,
+                evm_inquirer=evm_inquirer,
+                encounter=TokenEncounterInfo(
+                    description='Querying gearbox pools',
+                    should_notify=False,
+                ),
+            )
+        except NotERC20Conformant as e:
+            log.error(
+                f'Skipping pool {pool} because {pool.underlying_token} is not a '
+                f'valid ERC20 token. {e}',
+            )
+            continue
+
+        # ensure the farming tokens exist in the db
+        verified_farming_tokens: set[str] = set()
+        for token_address in pool.farming_pool_addresses:
+            try:
+                token = get_or_create_evm_token(
+                    userdb=evm_inquirer.database,
+                    evm_address=string_to_evm_address(token_address),
+                    chain_id=evm_inquirer.chain_id,
+                    evm_inquirer=evm_inquirer,
+                    underlying_tokens=[UnderlyingToken(
+                        address=pool.underlying_token,
+                        token_kind=EvmTokenKind.ERC20,
+                        weight=ONE,
+                    )],
+                    encounter=TokenEncounterInfo(
+                        description='Querying gearbox pools',
+                        should_notify=False,
+                    ),
+                )
```

Comment:
this logic is duplicated, can you merge both of the loops? You can do a `zip(('pools', 'farm'), (pools, farms))`